File: golang/lib/services/shared_file_object/shared_file_object.go
package shared_file_object

//TODO add documentation
type SharedFileObject struct {
	//The file absolute path in the container where this code is running
	absFilepathOnThisContainer string
	//The file absolute path in the service container
	absFilepathOnServiceContainer string
}

func NewSharedFileObject(absFilepathOnThisContainer string, absFilepathOnServiceContainer string) *SharedFileObject {
	return &SharedFileObject{absFilepathOnThisContainer: absFilepathOnThisContainer, absFilepathOnServiceContainer: absFilepathOnServiceContainer}
}

//TODO add documentation
func (s SharedFileObject) GetAbsFilepathOnThisContainer() string {
	return s.absFilepathOnThisContainer
}

//TODO add documentation
func (s SharedFileObject) GetAbsFilepathOnServiceContainer() string {
	return s.absFilepathOnServiceContainer
}

File: typescript/src/lib/services/shared_file_object/shared_file_object.ts
//TODO add documentation
export class SharedFileObject {
    //The file absolute path in the container where this code is running
    private readonly absFilepathOnThisContainer: string
    //The file absolute path in the service container
    private readonly absFilepathOnServiceContainer: string

    constructor(absFilepathOnThisContainer: string, absFilepathOnServiceContainer:string) {
        this.absFilepathOnThisContainer = absFilepathOnThisContainer;
        this.absFilepathOnServiceContainer = absFilepathOnServiceContainer
    }
    //TODO add documentation
    public getAbsFilepathOnThisContainer(): string {
        return this.absFilepathOnThisContainer
    }
    //TODO add documentation
    public getAbsFilepathOnServiceContainer(): string {
        return this.absFilepathOnServiceContainer
    }
}
